1_fetch/src/get_wqp_inventory.R

```diff
@@ -156,17 +156,15 @@ subset_inventory <- function(wqp_inventory, aoi_sf, buffer_dist_m = 0){
     sf::st_filter(y = sf::st_transform(aoi_sf, sf::st_crs(.)),
                   .predicate = st_is_within_distance,
                   dist = units::set_units(buffer_dist_m, m)) %>%
+    mutate(lon = sf::st_coordinates(.)[,1],
+           lat = sf::st_coordinates(.)[,2]) %>%
     sf::st_drop_geometry() %>%
-    pull(MonitoringLocationIdentifier) %>%
-    unique()
```

Comment:
Is there not a chance that we could still have duplicates since you start by using the inventory, where each site may appear more than once?

Reply:
The prior approach would have kept any duplicates too since we were just filtering the wqp inventory by sites that appeared in the list of unique strings from `queried_sites_aoi$MonitoringLocationIdentifier`.

One reason I'm not too concerned about this is because any duplicates in `p1_wqp_inventory_aoi` would get summarized in `p1_site_counts` as unique identifier-lat-lon-datum-grid sets. But it's possible I'm not thinking of edge cases that might crop up..

Reply:
> I'm not too concerned about this is because any duplicates in p1_wqp_inventory_aoi would get summarized in p1_site_counts as unique identifier-lat-lon-datum-grid sets

This is helpful, thanks. Definitely agree that in terms of the actual data and information used, your downstream steps should take care of it. 

My last sticking point is that with `{targets}` looking for any differences to kick off downstream steps, I try to whittle down each step to only the data/info that are actually needed. Say this step has some duplicated rows, and we add a few more of those the next time around. `{targets}` would see that as a change to the output and initiate builds of downstream steps when it perhaps doesn't need to. It probably would stop quickly because you have that next step ready, but those are the types of things that can start to slow down pipelines and make you nervous when pieces start to rebuild that you didn't think would.

Reply:
That's an excellent point! Even if duplicates would be handled in `p1_site_counts`, there are really strong reasons for wanting to avoid any extra rows here that would cause `p1_wqp_inventory_aoi` to become outdated unnecessarily.

I think that in this case, we actually want all of the duplicates that are in this data frame. Any sites that are duplicated in `queried_sites_aoi` are duplicated because that site has data for more than 1 of the characteristic names that we requested. Here's an example:

```r
> tar_load(p1_wqp_inventory)
> wqp_inventory <- p1_wqp_inventory
> 
> # Replicate steps within subset_inventory()
> queried_sites_transformed <- transform_site_locations(wqp_inventory, crs_out= "WGS84") 
Attempting to harmonize different site CRS...
> queried_sites_transformed_sf <- sf::st_as_sf(queried_sites_transformed, 
+                                              coords = c("lon","lat"), crs = 4326) 
> queried_sites_aoi <- queried_sites_transformed_sf %>%
+     sf::st_filter(y = sf::st_transform(aoi_sf, sf::st_crs(.)),
+                   .predicate = st_is_within_distance,
+                   dist = units::set_units(buffer_dist_m, m)) %>%
+     mutate(lon = sf::st_coordinates(.)[,1],
+            lat = sf::st_coordinates(.)[,2]) %>%
+     sf::st_drop_geometry() %>%
+     select(c(any_of(names(wqp_inventory)), datum))
> 
> # Now look for duplicated site ids
> queried_sites_aoi %>%
+     group_by(MonitoringLocationIdentifier) %>%
+     mutate(n_dups = n()) %>%
+     filter(n_dups > 1) %>%
+     select(MonitoringLocationIdentifier, activityCount, resultCount,
+            CharacteristicName, grid_id, n_dups) %>%
+     head()
# A tibble: 6 x 6
# Groups:   MonitoringLocationIdentifier [3]
  MonitoringLocationIdentifier activityCount resultCount CharacteristicName  grid_id n_dups
  <chr>                                <dbl>       <dbl> <chr>               <chr>    <int>
1 IL_EPA_WQX-RNZD-1                       27          31 Temperature, sample 11386        2
2 IL_EPA_WQX-RNZD-1                       33          33 Temperature, water  11386        2
3 IL_EPA_WQX-SDC-1                        48          48 Temperature, sample 11565        2
4 IL_EPA_WQX-SDC-2                        18          18 Temperature, sample 11565        2
5 IL_EPA_WQX-SDC-1                        23          23 Temperature, water  11565        2
6 IL_EPA_WQX-SDC-2                        20          20 Temperature, water  11565        2
> 
```

I lean towards keeping this information because this function/target are meant to create our inventory - i.e., which sites are there and what data exists for each of those sites. Would you agree with that?

Reply:
OK, you got me :) Thanks for laying it out with all these details; really helpful to wrap my brain around the problem space. I agree, this is a scenario where we do want these to kick off a build downstream.

Reply:
Thanks for your due diligence here! I appreciate your help thinking through the implications of these changes, and this thread was really helpful for me to think about kicking off builds only when we really mean to.